cpp/velox/substrait/SubstraitToVeloxPlan.cc

```diff
@@ -1543,47 +1521,63 @@ void SubstraitToVeloxPlanConverter::setFilterMap(
       if (substraitLit) {
         val = variant(substraitLit.value().i32());
       }
-      setColInfoMap<int>(functionName, colIdxVal, val, reverse, colInfoMap);
+      setColumnFilterInfo(functionName, colIdxVal, val, reverse, column2FilterInfo);
       break;
     case TypeKind::BIGINT:
       if (substraitLit) {
         if (inputType->isShortDecimal()) {
           auto decimal = substraitLit.value().decimal().value();
-          auto precision = substraitLit.value().decimal().precision();
-          auto scale = substraitLit.value().decimal().scale();
+          // auto precision = substraitLit.value().decimal().precision();
+          // auto scale = substraitLit.value().decimal().scale();
           int128_t decimalValue;
           memcpy(&decimalValue, decimal.c_str(), 16);
-          auto type = DECIMAL(precision, scale);
+          // auto type = DECIMAL(precision, scale);
           val = variant(static_cast<int64_t>(decimalValue));
         } else {
           val = variant(substraitLit.value().i64());
         }
       }
-      setColInfoMap<int64_t>(functionName, colIdxVal, val, reverse, colInfoMap);
+      setColumnFilterInfo(functionName, colIdxVal, val, reverse, column2FilterInfo);
       break;
     case TypeKind::DOUBLE:
       if (substraitLit) {
         val = variant(substraitLit.value().fp64());
       }
-      setColInfoMap<double>(functionName, colIdxVal, val, reverse, colInfoMap);
+      setColumnFilterInfo(functionName, colIdxVal, val, reverse, column2FilterInfo);
       break;
     case TypeKind::BOOLEAN:
       if (substraitLit) {
         val = variant(substraitLit.value().boolean());
       }
-      setColInfoMap<bool>(functionName, colIdxVal, val, reverse, colInfoMap);
+      setColumnFilterInfo(functionName, colIdxVal, val, reverse, column2FilterInfo);
       break;
     case TypeKind::VARCHAR:
       if (substraitLit) {
         val = variant(substraitLit.value().string());
       }
-      setColInfoMap<std::string>(functionName, colIdxVal, val, reverse, colInfoMap);
+      setColumnFilterInfo(functionName, colIdxVal, val, reverse, column2FilterInfo);
       break;
     case TypeKind::DATE:
       if (substraitLit) {
         val = variant(Date(substraitLit.value().date()));
       }
-      setColInfoMap<int>(functionName, colIdxVal, val, reverse, colInfoMap);
+      setColumnFilterInfo(functionName, colIdxVal, val, reverse, column2FilterInfo);
+      break;
+    case TypeKind::HUGEINT:
```

Comment:
Is it possible to add the support for filter pushdown of int128_t in another PR, and focus on code refactor in this one?

Reply:
ok